lib/components.js

```diff
@@ -6,6 +6,231 @@
+         * Each element that needs this check implements it itself.
+         * @returns {boolean}
+         */
+        static rangeUnderflow() {
+            return false;
+        }
+
+        /**
+         * Check if element is required and its value is missing
+         * @param {HTMLElement} element
+         * @returns {boolean}
+         */
+        static valueMissing(element) {
+            return element.hasAttribute('required') && !element.value;
+        }
+
+        /**
+         * Check if element name is missing
+         * @param {HTMLElement} element
+         * @returns {boolean}
+         */
+        static nameMissing(element) {
+            return !element.name && !element.getAttribute('name');
+        }
+
+        /**
+         * Check if an element is required
+         * @param {HTMLElement} element
+         * @returns {boolean}
+        */
+        static isRequired(element) {
+            return element.hasAttribute('required');
+        }
+
+        static customError() {
+            return false;
+        }
+
+        /**
+         * Checks if element is going to be serialized.
+         * If an element doesn't have a name it will not be serialized.
+         * Used to determine if an element should be validated.
+         * @param {HTMLElement} element
+         * @returns {boolean}
+        */
+        static willSerialize(element) {
+            return this.nameMissing(element) ? false : true;
+        }
+    }
+
+    /**
+     * The NativeElementValidator uses the methods from the Validator class
+     * All native elements tha don't support methods like isFormElement, tooLong, tooShort
+     * etc.. will be wrapped in this class in order to enable us to validate native and
+     * custom elements using the same methods.
+     * */
+    class NativeElementValidator {
+        constructor(element) {
+            this.element = element;
+        }
+
+        isFormElement() {
+            return Validator.isFormElement(this.element);
+        }
+
+        /**
+         * Check if element value is bigger than element maximum
+         * @returns {boolean}
+         */
+        tooLong() {
+            const max = parseFloat(this.element.getAttribute('maximum'));
+            if (!max) return false;
+            return this.element.value.length > max;
+        }
+
+        /**
+         * Check if element value is less than element minimum
+         * @returns {boolean}
+         */
+        tooShort() {
+            const min = parseFloat(this.element.getAttribute('minimum'));
+            if (!min) return false;
+            return this.element.value.length < min;
+        }
+
+        /**
+         * Checks if the value of an element is bigger than its max attribute
+         * @returns {boolean}
+        */
+        rangeOverflow() {
+            const max = parseFloat(this.element.getAttribute('max'));
+            if (!max) return false;
+            return parseFloat(this.element.value) > max;

```

Comment:
` if (!max)` will emit false if max is NaN which is the return value of parseFloat if the passed argument is an empty string.
I don't see a need to check for empty string except to avoid one parseFloat call. Do you think the perf gain is significant in this case? Maybe if you're updating a slider like a healthbar and you change its value too frequently. 

Changed.
Also changed in rangeUnderflow, tooLong and tooShort.

Reply:
Yes, we've ignored/looked pass `parseFloat`.